### PR TITLE
AES GCM Stream changes

### PR DESCRIPTION
**Pull request by ggershinsky**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/encryption/AesGcmInputStream.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public int read(byte[] b, int off, int len) throws IOException {
     }
 
     if (available() <= 0 && len > 0) {
-      throw new EOFException();
+      return -1;
```
Comment (ggershinsky) [inline core/src/main/java/org/apache/iceberg/encryption/AesGcmInputStream.java:115]: https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-int-int-

File: core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java [hunk 2]
```diff
@@ -95,6 +102,10 @@ public void write(byte[] b, int off, int len) throws IOException {
 
   @Override
   public long getPos() throws IOException {
```
Comment (ggershinsky) [inline core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java:104]: Iceberg manifest writer calls getPos after closing the avro stream. Wrong value was returned.

File: core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java [hunk 3]
```diff
@@ -56,6 +58,7 @@ public class AesGcmOutputStream extends PositionOutputStream {
     this.currentBlockIndex = 0;
     this.isHeaderWritten = false;
     this.lastBlockWritten = false;
+    this.isClosed = false;
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java:61]: Nit: initialize `finalPosition` as well?

Comment (ggershinsky) [reply core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java:61]: Done.

**Review by rdblue — APPROVED** [review]

**Pull request merged**